### PR TITLE
Add a reindex argument to the PDBWritter

### LINKED ISSUE #1072: Add option to keep original Atom IDs when writing PDB
### Expected behaviour

When I use `ag.write('foo.pdb')` I expect the atom ids to be the same as in the AtomGroup

### Actual behaviour

The atom ids's always start from 0.

### Code to reproduce the behaviour

``` python
import MDAnalysis as mda
from MDAnalysisTests.datafiles import PDB

u = mda.Universe(PDB)
bb = u.select_atoms('backbone')
bb.write('foo.pdb')
u2 = mda.Universe('foo.pdb')
assert np.all(bb.ids == u2.atoms.ids)
```

### Currently version of MDAnalysis:
(run `python -c "import MDAnalysis as mda; print(mda.__version__)"`)
develop branch

### PR DESCRIPTION
**Pull request by jbarnoud**
Fixes #1072

Changes made in this Pull Request:
 - Add a `reindex` argument to the `PDBWriter`. The argument is `True` by default and the atom serials are generated starting from 1. If the argument is set to `False`, then `atoms.ids` is used for the serial. This is reminiscent of #1781.

PR Checklist
------------
 - [x] Tests?
 - [X] Docs?
 - [x] CHANGELOG updated?
 - [X] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/coordinates/PDB.py [hunk 1]
```diff
@@ -533,6 +533,9 @@ class PDBWriter(base.WriterBase):
        ChainID now comes from the last character of segid, as stated in the documentation.
        An indexing issue meant it previously used the first charater (Issue #2224)
 
+    .. versionchanged:: 2.0.0
+        Add the "reindex" argument.
```
Comment (orbeckst) [inline package/MDAnalysis/coordinates/PDB.py:537]: State which value of `reindex` preserves old (pre 2.0.0) behavior.

Comment (orbeckst) [reply package/MDAnalysis/coordinates/PDB.py:537]: I think we put kwargs as bare reST roles i.e.
```reST
Add the `redindex` argument. Setting this keyword to ``True`` (the default) preserves
the behavior in earlier versions of MDAnalysis.
```

File: package/MDAnalysis/coordinates/PDB.py [hunk 2]
```diff
@@ -624,6 +627,9 @@ def __init__(self, filename, bonds="conect", n_atoms=None, start=0, step=1,
            ``False``: write a single frame to the file; ``True``: create a
            multi frame PDB file in which frames are written as MODEL_ ... ENDMDL_
            records. If ``None``, then the class default is chosen.    [``None``]
+        reindex: bool (optional)
+            If ``True`` (default), the atom serial is set to be consecutive
```
Comment (orbeckst) [inline package/MDAnalysis/coordinates/PDB.py:631]: 1. Which value preserves compatibility with 1.x?
2. Have we ever been able to rountrip a PDB when the serials were not consecutively numbered starting from 1? Is this something we would want to be able to do? Is this something a user would generally expect?

**Review by orbeckst — COMMENTED** [review]
Only some half-backed comments, sorry, not much time.

Does the PDB standard prescribe continuous serials starting from 1? (IIRC we never really found out but maybe someone knows.) 

What is the user's expectation: save PDB with exactly the given serials or save with something looking like a "normal" PDB?

If users want renumbered serials, should we have them renumber a serial attribute in the topology instead and then use that for writing instead of just doing something on the fly?

**Comment by jbarnoud** [comment]
What we called the atom id is actually called atom serial, which let me assume it should be consecutive and starting at 1. In the discussion in the issue, @arose commented in that direction. So the default seems to be what is expected. However, it is unconvenient for some usecase, especially when patching systems together; hence the new keyword.

**Comment by orbeckst** [comment]
Thanks for the explanation. Then this makes sense, with the default the most likely PDB-standard implementation and `reindex=False` for the special cases.

**Review by orbeckst — CHANGES_REQUESTED** [review]
I am happy with the UI. Perhaps someone else can have a look at the implementation.

The test coverage is still a bit low, so that's my only complaint.

**Comment by jbarnoud** [comment]
@orbeckst 

> The test coverage is still a bit low, so that's my only complaint.

With 100% patch coverage, it should be good now.

**Review by orbeckst — APPROVED** [review]

**Comment by orbeckst** [comment]
Thanks, could you please address the merge conflict in CHANGELOG? Otherwise all good from my side.

**Pull request merged**